tests/integration/targets/ec2_spot_instance/tasks/main.yaml

```diff
@@ -230,6 +277,19 @@
         ignore_errors: true
         retries: 5
 
+      - name: Get info about the complex spot instance request created
```

Comment:
In the `always` block we want to use the module a little differently.  We shouldn't really be testing here, this is just for cleanup.  But we would like to make use of the info module to remove the hard-coded variables at lines 275 and 275.
```
            - '{{ create_result.spot_request.spot_instance_request_id }}'
            - '{{ complex_create_result.spot_request.spot_instance_request_id }}'
```
Instead we'd add a `ec2_spot_instance_info` task before the`remove spot instance requests` task that filters on vpc-id `vpc_result.vpc.id`.  The result from that should be every single sir we created in the test, whether that's 1, 2, or 20 of them.  Then in the delete task, we can have a `with_items` to loop through.   This way if the test suite grows in the future, we can be sure we're always deleting everything without having to keep adding static lines to the cleanup task.   That would look something like:

```
      - name: remove spot instance requests
        ec2_spot_instance:
          spot_instance_request_ids:
            - '{{ item.spot_instance_request_id }}'
          state: 'absent'
        ignore_errors: true
        retries: 5
        with_items: '{{ var_you_registered_in_info_task.spot_request }}'
```
There's a pretty good example here https://github.com/ansible-collections/amazon.aws/blob/main/tests/integration/targets/ec2_instance/roles/ec2_instance/tasks/env_cleanup.yml#L10-L24